### PR TITLE
Add simplifierx switch

### PR DESCRIPTION
**Pull request by dopamane**
Fixes https://github.com/kframework/kore/issues/2939

See #2923 for simplifier discussion.

### Scope:
* Add `--simplifierx` switch

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/app/share/GlobalMain.hs [hunk 1]
```diff
@@ -270,6 +269,8 @@ the project
 data GlobalOptions = GlobalOptions
     { -- | Version flag [default=false]
       willVersion :: !Bool
+    , -- | Experimental simplifier flag [default=false]
+      simplifierx :: !Bool
```
Comment (ana-pantilie) [inline kore/app/share/GlobalMain.hs:273]: I'd rather create a new datatype instead of using `Bool`, `data ExperimentalSimplifier = Enabled | Disabled`.

**Review by ana-pantilie — CHANGES_REQUESTED** [review]
This only parses an additional CL option, but doesn't do anything with it. This needs to be threaded to the simplifier.

A field of the `ExperimentalSimplifier` type needs to be added to the [simplifier's environment](https://github.com/kframework/kore/blob/dd3d4d9da6d197215e7e6cef7ca235ecfcedc709/kore/src/Kore/Simplify/Data.hs#L84). When the simplifier is ran, it needs to be set to the option received from the command line.

You should also add a method for extracting the `ExperimentalSimplifier` to the [Simplifier class](https://github.com/kframework/kore/blob/dd3d4d9da6d197215e7e6cef7ca235ecfcedc709/kore/src/Kore/Simplify/Simplify.hs#L148). This way, it's available throughout all the simplification code.

That's just the first part. The second part is about making changes in our test suites to accommodate the new option.

The integration tests require changes made to the Makefiles, we should probably add a new target like `make testx` which runs the tests with the `simplifierx` option turned on. This also needs to be added on CI.

In the case of the unit tests, the changes to the simplifier will trigger some compilation errors wherever some form of `run/eval/...simplifier` is defined in the tests. We should see if it's possible to change these so that it runs the simplifier twice: once with the switch turned off and once with the switch turned on.

It remains to be discussed if we think it's necessary to run all of our tests twice.

In the case of the unit tests, things will most likely break on the experimental branch. This is because many unit tests are old and very general. This is good because we can explicitly see what features we will be breaking and make decisions based on concrete examples, but also bad because it will be more time consuming.

In the case of the integration tests, we definitely need to get everything working on the experimental branch. The problem is that we will suffer from much longer CI times. Therefore, I think this PR should not enable CI with the switch turned on just yet, it should add the necessary infrastructure so that when we add the first experimental feature we can turn it on easily.

**Comment by ana-pantilie** [comment]
Since I need new work, I picked this up. I tried to work with having the new option added to `GlobalOptions`, but it seems to me like these options were added to only be used in `GlobalMain`, before calling the specific executable. Since I can't really use your changes, I'm going to open a new PR. That will also mean that you can review/approve it when it's ready.

**Comment by dopamane** [comment]
Ok, thank you for the explanation!

**Pull request closed without merging**